src/aux_utils.cpp

```diff
@@ -828,6 +864,15 @@ namespace diskann {
 
     diskann_writer.write(sector_buf.get(), SECTOR_LEN);
 
+    
+    unsigned  dummyUnsigned, loc_id;
+    std::unique_ptr<char[]> dummy_buf = std::make_unique<char[]>(max_node_len);
```

Comment:
if this dummy_buf is only there for moving the pointer, we can replace this with seekg in the loop, no?

Reply:
dummy_buf variable has been removed as it is not necessary.